src/wasm-lib/kcl/src/coredump/local.rs

```diff
@@ -0,0 +1,45 @@
+//! Functions for getting core dump information via local rust.
+
+use anyhow::Result;
+
+use crate::coredump::CoreDump;
+
+#[derive(Debug, Clone)]
+pub struct CoreDumper {}
+
+impl CoreDumper {
+    pub fn new() -> Self {
+        CoreDumper {}
+    }
+}
+
+impl Default for CoreDumper {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[async_trait::async_trait]
```

Comment:
ooh we could try using the new built-in async trait stuff.

Reply:
@adamchalmers @jessfraz it's not a bad idea to keep this for now; fwiw -- async traits as implemented in the stdlib are not object safe, which means we can't ever have a `&Trait` or a `Trait` varaible, we'd only be able to use non-async_trait async traits as generic type restriction and shit. It's really fucking annoying and got me bad. Let me write some snippits real quick, but i'm inclined to keep using `async_trait` until someone figures out object safety with async rust

Reply:
```rust
use std::future::Future;

trait NotObjectSafe {
    async fn foo(&self);
}

trait NotObjectSafeDesugar {
    fn foo(&self) -> impl Future<Output=()>;
}

// the trait `NotObjectSafe` cannot be made into an object
fn foo(x: &NotObjectSafe) {}

// allowed, since `F` isn't a trait, it has a concrete type
// that is object-safe, it just has the trait `NotObjectSafe`
fn foo<F>(x: F) where F: NotObjectSafe {}

fn main() {
    // Compiler error! This isn't valid.
    // trait `NotObjectSafe` cannot be made into an object
    let foo: NotObjectSafe;
    
    // Compiler error! This isn't valid.
    // trait `NotObjectSafe` cannot be made into an object
    let foo: NotObjectSafeDesugar;
}
```

Reply:
(relatedly; this is very very very annoying)